src/Extension/CoreExtension.php

```diff
@@ -1595,4 +1595,25 @@ function twig_array_reduce($array, $arrow, $initial = null)
 
     return array_reduce($array, $arrow, $initial);
 }
+
+/**
+ * Returns the modulo for floating point numbers
+ *
+ * @param int|float $divident  The dividend
+ * @param int|float $divisor The divisor
+ *
+ * @return int|float
+ */
+function twig_mod($divident, $divisor)
+{
+    if (is_int($divident) && is_int($divisor)) {
+        return $divident % $divisor;
+    }
+
+    $dividentDecimal = ($pos = strpos($divident, '.')) ? strlen($divident) - $pos - 1 : 0;
+    $divisorDecimal = ($pos = strpos($divisor, '.')) ? strlen($divisor) - $pos - 1 : 0;
```

Comment:
you should not do it using string manipulation (especially given that float casting is locale-dependant before PHP 8 so this is not even working if the locale uses a different decimal separator).